apps/ecosystem/src/ecosystem.repository.ts

```diff
@@ -1369,4 +1369,134 @@ export class EcosystemRepository {
       throw error;
     }
   }
-}
+
+  async getEcosystemsByOrgId(orgId: string): Promise<IEcosystemData[]> {
+    try {
+
+      const ecosystemsDetails = await this.prisma.ecosystem.findMany({
+        where: {
+          ecosystemOrgs: {
+            some: {
+              orgId
+            }
+          }
+          
+        },
+        include: {
+          ecosystemOrgs: {
+            include: {
+              ecosystemRole: true
+            }
+          }
+        }
+      });
+      
+      return ecosystemsDetails;
+    } catch (error) {
+      this.logger.error(`Error in getting ecosystems: ${error.message}`);
+      throw error;
+    }
+  }
+
+  async getOrgName(orgId: string): Promise<{
+    name: string;
+  }> {
+    try {
+      const orgName = await this.prisma.organisation.findUnique({
+        where: {
+          id: orgId
+        },
+        select: {
+          name: true
+        }
+      });
+
+      return orgName;
+    } catch (error) {
+      this.logger.error(`Error in getting organization names: ${error.message}`);
+      throw error;
+    }
+  }
+
+  async getEcosystemLeadOrgDetails(orgId: string): Promise<{
+    name: string;
+  }> {
+    try {
+      const orgName = await this.prisma.organisation.findUnique({
+        where: {
+          id: orgId
+        },
+        select: {
+          name: true
+        }
+      });
+
+      return orgName;
+    } catch (error) {
+      this.logger.error(`Error in getting organization names: ${error.message}`);
+      throw error;
+    }
+  }
+
+  async deleteEcosystemInvitations(orgId: string): Promise<Prisma.BatchPayload> {
+    try {
+      const deletedEcosystemInvitations = await this.prisma.ecosystem_invitations.deleteMany({
+        where: {
+          orgId
+        }
+      });
+
+      return deletedEcosystemInvitations;
+    } catch (error) {
+      this.logger.error(`Error in deleting ecosystem invitations: ${error.message}`);
+      throw error;
+    }
+  }
+
+  async deleteEcosystems(orgId: string): Promise<IEcosystemDataDeletionResults> {
+    try {
+      return await this.prisma.$transaction(async (prisma) => {
+        const deletedEcosystemUsers = await prisma.ecosystem_users.deleteMany({
+          where: {
+            ecosystem: {
+              ecosystemOrgs: {
+                some: {
+                  orgId
+                }
+              }
+            }
+          }
+        });
+
+        const deleteEndorsementTransactions = await prisma.endorsement_transaction.deleteMany({
```

Comment:
We are directly calling the endorsement_transaction table from the ecosystem MS

Reply:
endorsement_transaction table is the part of ecosystem microservice. We are using endorsement_transaction table for ecosystem flow